trieste/acquisition/function.py

```diff
@@ -512,15 +501,158 @@ def prepare_acquisition_function(
         return lambda at: expected_improvement(objective_model, eta, at) * constraint_fn(at)
 
 
-class BatchAcquisitionFunctionBuilder(ABC):
-    """ A :class:`BatchAcquisitionFunctionBuilder` builds a batch acquisition function. """
+def independent_latent_sampler_with_reparametrisation(
+    dataset: Dataset, model: ProbabilisticModel, sample_size: int, observation_shape: ShapeLike,
+) -> Callable[[tf.Tensor], tf.Tensor]:
+    """
+    Returns independent samples according to the reparametrization trick.
+    The sample size S is determined by eps,
+    N is the number of query points and L is the number of latent processes.
+    :param dataset:
+    :param model:
+    :param sample_size:
+    :param observation_shape:
+    :return: a [S, N, L] tensor
+    """
+    eps = tf.random.normal([sample_size] + observation_shape, dtype=dataset.observations.dtype)
 
-    @abstractmethod
-    def prepare_acquisition_function(
-        self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
-    ) -> BatchAcquisitionFunction:
+    def sample(at: tf.Tensor) -> tf.Tensor:
+        tf.debugging.assert_shapes([at, 'ND'])
+        mean, cov = model.predict(at)   # [N, L], [N, L]
+        return mean + tf.math.sqrt(cov) * eps[:, None, :]  # [S, N, L]
+
+    return sample
+
+
+# todo the batch size problem
+def batch_latent_sampler_with_reparametrisation(
+    dataset: Dataset,
+    model: ProbabilisticModel,
```

Comment:
how about the case of multiple models? cf inequality notebook

Reply:
agreed to do in another PR